proposals/p4864.md

```diff
@@ -0,0 +1,155 @@
+# No predeclared identifiers, `Core` is a keyword
+
+<!--
+Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+Exceptions. See /LICENSE for license information.
+SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+-->
+
+[Pull request](https://github.com/carbon-language/carbon-lang/pull/4864)
+
+<!-- toc -->
+
+## Table of contents
+
+-   [Abstract](#abstract)
+-   [Problem](#problem)
+-   [Background](#background)
+-   [Proposal](#proposal)
+-   [Details](#details)
+-   [Rationale](#rationale)
+-   [Future work](#future-work)
+    -   [Package name `Cpp`](#package-name-cpp)
+    -   [Package name `Main`](#package-name-main)
+-   [Alternatives considered](#alternatives-considered)
+    -   [Have both predeclared identifiers and keywords](#have-both-predeclared-identifiers-and-keywords)
+    -   [Reserve words with a certain spelling](#reserve-words-with-a-certain-spelling)
+
+<!-- tocstop -->
+
+## Abstract
+
+Introduce a principle that the Carbon language should not encroach on the
+developer's namespace. Satisfy this principle by making `Core` a keyword.
+
+## Problem
+
+Ongoing design work needs rules for how to expose types such as a primitive
+array type to Carbon code, and in particular, if we choose to make it available
+by default, whether that should be accomplished by a keyword or a predeclared
+identifier.
+
+## Background
+
+See the
+[Background section of the added principle](/docs/project/principles/namespace_cleanliness.md#background).
+
+## Proposal
+
+We choose to not have any predeclared identifiers in Carbon. If a word has
+special meaning to the language, then that word is a keyword, and a plain
+identifier with no special meaning is always available using raw identifier
+syntax.
+
+## Details
+
+See [the principle document](/docs/project/principles/namespace_cleanliness.md)
+for details of the added principle. In addition, we make one change and one
+clarification:
+
+-   `Core` is changed from being an identifier that happens to be the name of
```

Comment:
Worth mentioning here and/or in the problem statement that this also removes ambiguity about the semantics when language features delegate to "something in the `Core` package" -> that cannot refer to any arbitrary name lookup spelled `Core`?

Reply:
Done here and in rationale.